app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -233,4 +292,37 @@ def extract_ems_ref_id(href)
       href && href.split("/").last
     end
   end
+
+  class Cacher
```

Comment:
@chrisarcand can you have a look at this? I wonder if a) we should/do use the `Rails.cache` and b) if this does not rebuild functionality thats probably already in `Rails.cache`